assembl/static2/js/app/components/common/visibilityComponent.jsx

```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+
+class VisibilityComponent extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
```

Comment:
My knowledge of React is still quite limited so for sure I don't use the best solutions. But I had already tried to use only props instead of state for this component, and it was not refreshing when its props changed. Also a constraint I have is that its parent component (ideasLevel) should not re-render() in the middle of an animation otherwise animation is cut, so I can't just lift the state up. This makes things difficult. I would welcome a pair programming session on this.